lib/open-amp/resource_table.c

```diff
@@ -29,7 +29,18 @@
 #include <zephyr/kernel.h>
 #include <resource_table.h>
 
-extern char ram_console[];
+#ifdef CONFIG_RAM_CONSOLE_BUFFER_SECTION
+#if DT_HAS_CHOSEN(zephyr_ram_console)
+
+static const uint32_t ram_console_buf = DT_REG_ADDR(DT_CHOSEN(zephyr_ram_console));
+
+#else
+#error "Lack of chosen property zephyr,ram_console!"
+#endif
+
```

Comment:
Is'nt it sufficient to just use the extern to get the already defined `ram_console_buf`. This has the `RAM_CONOLE_BUF_ATTR` defined for it which should place it in the appropriate RAM section.

Reply:
The ram_console_buf in ram_console.c is only used when CONFIG_RAM_CONSOLE_BUFFER_SECTION is disabled. That's why I changed it in ram_console.c to not wasting memory.